File: android/app/src/main/java/com/HammersTech/RoutineChart/core/domain/repositories/ChildProfileRepository.kt
package com.HammersTech.RoutineChart.core.domain.repositories

import com.HammersTech.RoutineChart.core.domain.models.ChildProfile
import kotlinx.coroutines.flow.Flow

/**
 * Repository interface for ChildProfile operations
 */
interface ChildProfileRepository {
    suspend fun create(profile: ChildProfile)
    suspend fun update(profile: ChildProfile)
    suspend fun getById(id: String): ChildProfile?
    fun observeById(id: String): Flow<ChildProfile?>
    suspend fun getByFamilyId(familyId: String): List<ChildProfile>
    fun observeByFamilyId(familyId: String): Flow<List<ChildProfile>>
}

File: ios/RoutineChart/RoutineChart/Core/Domain/Repositories/ChildProfileRepository.swift
//
//  ChildProfileRepository.swift
//  RoutineChart
//
//  Repository protocol for ChildProfile data access
//

import Foundation

protocol ChildProfileRepository {
    func create(_ childProfile: ChildProfile) async throws
    func get(id: String) async throws -> ChildProfile?
    func update(_ childProfile: ChildProfile) async throws
    func getAll(familyId: String) async throws -> [ChildProfile]
}

